pip/index.py

```diff
@@ -330,13 +335,16 @@ def mkurl_pypi_url(url):
                 ) or req.url_name
 
         if url_name is not None:
-            locations = [
-                mkurl_pypi_url(url)
-                for url in self.index_urls] + self.find_links
-        else:
-            locations = list(self.find_links)
+            return [mkurl_pypi_url(url) for url in self.index_urls]
+        return []
+
+    def _find_all_versions(self, req):
+        index_locations = self._get_index_urls_locations(req)
+        file_locations, url_locations = self._sort_locations(index_locations)
+        fl_file_loc, fl_url_loc = self._sort_locations(self.find_links)
```

Comment:
why change to 2 sort calls like this vs just  `self._sort_locations(index_locations + self.find_links)` like it was before?

I think the result is the same, but I had to do a double take to feel comfortable with that.